replace_mlrun.py

```diff
@@ -0,0 +1,133 @@
+        if self._verbose:
+            print(" ".join(cmd))
+        return subprocess.run(
+            cmd,
+            capture_output=True,
+            text=True,
+            check=True,
+        ).stdout
+
+    def _exec_silent_cmd(self, cmd: List[str]):
+        if self._verbose:
+            print(" ".join(cmd))
+        subprocess.run(cmd, stdout=subprocess.DEVNULL, check=True)
+
+    def _get_mlrun_api_pod(self) -> Union[str, None]:
+        print("Get mlrun-api pod name")
+        cmd = ["kubectl", "-n", "default-tenant", "get", "pod"]
+
+        for line in self._exec_cmd(cmd).splitlines()[1:]:
+            if "mlrun-api-chief" in line:
+                return line.split()[0]
+
+        return None
+
+    def _get_mlrun_api_image(self, pod_name: str) -> Union[str, None]:
+        print("Get mlrun-api image name")
+        cmd = ["kubectl", "-n", "default-tenant", "get", "pods", pod_name, "-o", "yaml"]
+
+        for line in self._exec_cmd(cmd).splitlines():
+            if "image" in line and "mlrun-api" in line:
+                return line.split("image:")[1].strip()
+
+        return None
+
+    def _build_api_docker(self, version: str) -> Union[str, None]:
+        os.environ["MLRUN_VERSION"] = version
+        image_tag_str = "Successfully tagged "
+
+        print("Build mlrun-api image")
+
+        cmd = ["make", "api"]
+
+        for line in self._exec_cmd(cmd).splitlines():
+            if image_tag_str in line and "mlrun-api" in line:
+                return line.split(image_tag_str)[-1].strip()
+        return None
+
+    def _replace_docker_tag(self, new_image: str, orig_image: str):
+        print("Replace docker tag")
+        cmd = ["docker", "tag", new_image, orig_image]
+        self._exec_silent_cmd(cmd)
+
+    def _delete_orig_pod(self):
+        print("Delete dockers, so that new images will replace them")
+        cmd = [
+            "kubectl",
+            "-n",
+            "default-tenant",
+            "delete",
+            "pod",
+            "-l",
+            "app.kubernetes.io/instance=mlrun",
+            "-l",
+            "app.kubernetes.io/component=api",
+        ]
+        self._exec_silent_cmd(cmd)
+
+    def do_replacing(self):
+        api_pod = self._get_mlrun_api_pod()
+        if not api_pod:
+            raise RuntimeError(
+                "mlrun-api pod name not found. perhaps you are running on old version without chief pod"
+            )
+
+        api_image = self._get_mlrun_api_image(api_pod)
+        if not api_image:
+            raise RuntimeError("mlrun-api image name not found")
+
+        api_image_version = api_image.split("mlrun-api:")[-1].strip()
+
+        if self._verbose:
+            print(f"Version: {api_image_version} from {api_image}")
+
+        new_image_name = self._build_api_docker(api_image_version)
+        if not new_image_name:
+            raise RuntimeError("unable to build api docker")
+
+        self._replace_docker_tag(new_image_name, api_image)
```

Comment:
Should we tag the old image as something else (`<orig>_bak` or something)? One might want to revert back to the original later.